documentation/asciidoc/topics/proc_creating_caches_templates.adoc

```diff
@@ -0,0 +1,50 @@
+[id='cache_templates-{context}']
+= Creating {brandname} Caches from Templates
+Complete the following steps to create caches on {datagridservice} clusters
+using `infinispan.org` cache configuration templates.
+
+.Prerequisites
+
+* Create an authentication secret that contains valid user credentials for
+accessing {brandname} clusters.
+* Identify the cache configuration template you want to use for your cache. You
+can retrieve all available configuration templates with the following REST call:
++
+[source,options="nowrap",subs=attributes+]
+----
+GET /rest/v2/cache-managers/default/cache-configs/templates
+----
+
+.Procedure
+
+. Create a **Cache** CR that specifies the name of the template you want to use.
++
+For example, the following CR creates a cache named "mycache" that uses the `org.infinispan.DIST_SYNC` cache configuration template:
++
+[source,options="nowrap",subs=attributes+]
+----
+apiVersion: infinispan.org/v2alpha1
+kind: Cache
+metadata:
+  name: mycachedefinition <1>
+spec:
+  adminAuth:
+    secretName: basic-auth <2>
+  clusterName: {example_crd_name} <3>
+  name: mycache <4>
+  templateName: org.infinispan.DIST_SYNC <5>
+----
++
+<1> Names the **Cache** CR.
+<2> Specifies the authentication secret with credentials for accessing {brandname} clusters.
```

Comment:
maybe we can specify the default value for keys in the secret? I mean `username` and `password`

Reply:
@rigazilla I'm not sure I follow. should we rephrase as "Specifies the username and password to access {brandname} clusters." ?

Reply:
@oraNod I mean if we need to describe the format of the credential secret: i.e. by default the operator is expecting to found the username in the secret entry with key `username` and the password in the secret entry with key `password`. User can override this default specifying a complete `adminAuth` field, like this:
```
    username:
      key: username     <- keyname for username in secret
      name: basic-auth <- secretname
    password:
      key: password    <- keyname for password in secret
      name: basic-auth <- secretname for password
```